app/users/views.py

```diff
@@ -0,0 +1,68 @@
+from typing import Any
+
+from django.contrib.auth import get_user_model
+from rest_framework import generics, response, status
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework_simplejwt.tokens import RefreshToken
+
+from app.users.serializers import UserSerializer
+
+from .serializers import PasswordResetSerializer, ResetPasswordSerializer
+
+User = get_user_model()
+
+
+class UserRegister(APIView):
+    def post(self, request: Request, format: str = "json") -> Response:
+        serializer = UserSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = serializer.save()
+        refresh = RefreshToken.for_user(user)
+        response = serializer.data
+        response["refresh"] = str(refresh)
+        response["access"] = str(refresh.access_token)
+
+        return Response(response, status=status.HTTP_201_CREATED)
+
+
+class PasswordReset(generics.GenericAPIView):
+    """
+    Request for Password Reset Link.
+
+    """
+
+    serializer_class = PasswordResetSerializer
+
+    def post(self, request: Request, format: str = "json") -> Response:
+
+        serializer = self.get_serializer(
+            data={"email": request.data.get("email")}
+        )
+        serializer.is_valid(raise_exception=True)
+        return Response(
+            {"message": "check your email for password reset link"},
+            status=status.HTTP_200_OK,
+        )
+
+
+class ResetPasswordAPI(generics.GenericAPIView):
+
+    serializer_class = ResetPasswordSerializer
+
+    def post(self, request: Any, *args: Any, **kwargs: Any) -> Any:
+        """
+        Verify token & encoded_pk and then reset the password.
+
+        """
+        serializer = self.serializer_class(
```

Comment:
use self.get_serializer instead of self.serializer_class

Reply:
What is the difference between the two ?